src/kolibri_android/main_activity/activity.py

```diff
@@ -94,6 +94,16 @@ def __init__(self):
             Runnable(self._on_loading_ready),
         )
 
+    def on_activity_stopped(self, activity):
+        super().on_activity_stopped(activity)
+        if self._kolibri_bus is not None:
+            self._kolibri_bus.transition("STOP")
```

Comment:
I was a little suspicious of this since it's transitioning to different target states than we use [in kolibri-daemon](https://github.com/learningequality/kolibri-installer-gnome/blob/master/src/kolibri_daemon/kolibri_http_process.py#L103-L122) and I remember that being troublesome, but it all checks out. I wonder if these state transitions were improved with 0.16? (Or I could just be remembering something else).

Reply:
Oh interesting. I didn't remember that you had to play with magicbus in kolibri-daemon as well. I will test what happens if change it to transition to the "IDLE" state.

Also in the playthrough yesterday we saw that the contentimport task doesn't obey to cancel as fast as channelimport. It could go for several seconds. So I think I'll need to add a waiter similar to what Kolibri 16 has around here https://github.com/learningequality/kolibri/blob/ecf08f1f07b3ae8c6597ccaf708698c8e918e032/kolibri/utils/server.py#L599.